task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -66,12 +68,15 @@ class RuntimeTaskInstance(TaskInstance):
     _ti_context_from_server: Annotated[TIRunContext | None, Field(repr=False)] = None
     """The Task Instance context from the API server, if any."""
 
-    def get_template_context(self):
+    max_tries: int = 0
```

Comment:
The other option is to instead just keep `max_tries` in the context dict.

Reply:
It is different than `BaseOperator.retries` because of clearing behavior:

https://github.com/apache/airflow/blob/1cf1d628404ab62f979d2b0d9936ca5af001f44f/airflow/models/taskinstance.py#L464-L468

Reply:
It is used in `BaseSensor` for reschedule:

https://github.com/apache/airflow/blob/1cf1d628404ab62f979d2b0d9936ca5af001f44f/airflow/sensors/base.py#L219-L229

Reply:
Captured some of it in https://github.com/apache/airflow/issues/45580